repository.go

```diff
@@ -275,6 +269,25 @@ func (r *Repository) Pull(o *PullOptions) error {
 	return r.createReferences(head)
 }
 
+// Push pushes changes to a remote.
+func (r *Repository) Push(o *PushOptions) error {
+	if o.RemoteName == "" {
+		o.RemoteName = DefaultRemoteName
+	}
+
+	remote, err := r.Remote(o.RemoteName)
+	if err != nil {
+		return err
+	}
+
+	if err := remote.ConnectPush(); err != nil {
```

Comment:
I think Connect should not be needed here, unless you want to do other things over the connection in addition to Push.